src/partcad/shape.py

```
@@ -918,13 +925,27 @@ async def _materialize_output_script(self, ctx, impl):
         if not impl.script:
+            if builtin_package is None:
+                # 'cae:' has no built-in package to fall back to, so an
+                # unresolved implementation means the configured one was not
+                # found rather than that somebody forgot a 'path'. Say which
+                # knob names it, since it is a user-configuration default rather
+                # than anything in the package being analysed.
+                raise Exception(
+                    "No implementation of '%s' is declared. Name one in a 'cae:' section, "
+                    "or set the default with 'pc cae %s --implementation <package>:<type>'"
+                    % (impl.format_name, impl.format_name)
+                )
```

Ruff (0.16.3)

[warning] Create your own exception

(TRY002)